### DISCUSSION TITLE
[Prettier Plugin] Make it possible to sort class names in arbitrary strings

### DISCUSSION THREAD

**Discussion by nikwen** [tailwindlabs/tailwindcss#10458] [Ideas]
Sometimes it is necessary to have class names in arbitrary strings. It would be great to be able to sort them using the Tailwind CSS Prettier plugin.

Examples:

```js
var className1 = 'sm:p-1 p-0';
let className2 = "sm:p-1 p-0";
const className3 = `sm:p-1 p-0`;

let obj1 = {
  className: "sm:p-1 p-0",
};

let obj2 = {
  ["className"]: Math.random() >= 0.5 ? "sm:p-1 p-0" : "sm:p-2 p-1",
};
```

This was also requested in https://github.com/tailwindlabs/prettier-plugin-tailwindcss/pull/97.

### Comments

By @jamescmartinez:

```js
// prettier-plugin-tailwindcss
const base = "mb-4 text-white"
```

### Via the Tailwind CSS Intellisense plugin

There was also a [discussion](https://github.com/tailwindlabs/prettier-plugin-tailwindcss/pull/97#issuecomment-1310829826) about implementing sorting functionality in the Tailwind CSS Intellisense plugin. I'd advocate against that because:

* A lot of devs use Jetbrains IDEs instead of VS Code (myself included).
* It cannot be automated using tools such as [husky](https://www.npmjs.com/package/husky) and [lint-staged](https://www.npmjs.com/package/lint-staged).

**Comment by joulev** [comment]
Agreed about the Intellisense plugin part. Although most if not all members in my team, me included, use VSCode, our workflow should not be made dependent on any specific editors.

**Comment by RoyRao2333** [comment]
Yes, I absolutely support this proposal. Sometimes we have some duplicated components and we only wanna take the **className** out as a variable. But then everytime you edit that variable, you can't be sure the order of those classes is right.

**Comment by Jorge-Luis-Rangel-Peralta** [comment]
I also think this would be useful
